src/webhook/pkg/mutating_webhook/remove-resourcespec.go

```diff
@@ -80,11 +81,12 @@ func adjustResourceSpec(pod *corev1.Pod) *corev1.Pod {
 
 	// Add cpu annotation
 	if !cpuRequest.IsZero() && cpuLimit.Cmp(cpuRequest) >= 0 {
-		logger.Printf("Adding CPU annotation based on cpuLimit: %s", cpuLimit.String())
-		annotations[PEERPODS_CPU_ANNOTATION] = cpuLimit.String()
+		logger.Printf("Adding CPU annotation based on cpuLimit (scaled value): %d", cpuLimit.Value())
```

Comment:
The documentation of https://pkg.go.dev/k8s.io/apimachinery/pkg/api/resource#Quantity.Value got me confused as it says `Value returns the unscaled value of q rounded up to the nearest integer away from 0`. I'm not expect here so just bringing to your attention to double-check we are using the right function.

Reply:
Basically the my comment may be misleading. I will remove the scale word. The key is rounding up to the nearest integer.

Reply:
ah ok, I was suspicious that we wanted rounding up instead :)

Reply:
kata and caa cannot handle vcpus values as 100m or 0.5, so we need it to be rounded to the neared integer :-)